db-engine/src/main/java/com/splicemachine/db/impl/sql/compile/UnaryDateTimestampOperatorNode.java

```diff
@@ -181,4 +183,12 @@ public void generateExpression(ExpressionClassBuilder acb,
         mb.cast(ClassName.DataValueDescriptor);
         mb.callMethod(VMOpcode.INVOKEINTERFACE,(String)null,methodName,getTypeCompiler().interfaceName(),1);
     } // end of generateExpression
+
+    @Override
+    public double getBaseOperationCost() throws StandardException {
+        double lowerCost = getOperandCost();
+        double localCost = SIMPLE_OP_COST * (operand == null ? 1.0 : 2.0);
```

Comment:
you have consts for everything but not for 1.0 and 2.0, and i'm not sure what they represent. Why is the cost twice if there's an operand?

Reply:
It just indicates that if the function has an operand, probably it executes more instructions to finish. I didn't look into the code of `date` and `timestamp`. But the costing is also like a heuristic. No need to make the cost absolutely accurate because in the end, it's about comparing options, always relative.